Mathlib/Topology/Category/ExtrDisc/ExplicitLimits.lean

```diff
@@ -0,0 +1,138 @@
+/-- The inclusion of one of the factors into the explicit finite coproduct. -/
+def finiteCoproduct.ι (a : α) : X a ⟶ finiteCoproduct X where
+  toFun := fun x => ⟨a,x⟩
+  continuous_toFun := continuous_sigmaMk (σ := fun a => X a)
+
+/--
+To construct a morphism from the explicit finite coproduct, it suffices to
+specify a morphism from each of its factors.
+This is essentially the universal property of the coproduct.
+-/
+def finiteCoproduct.desc {B : ExtrDisc.{u}} (e : (a : α) → (X a ⟶ B)) :
+    finiteCoproduct X ⟶ B where
+  toFun := fun ⟨a,x⟩ => e a x
+  continuous_toFun := by
+    apply continuous_sigma
+    intro a; exact (e a).continuous
+
+@[reassoc (attr := simp)]
+lemma finiteCoproduct.ι_desc {B : ExtrDisc.{u}} (e : (a : α) → (X a ⟶ B)) (a : α) :
+  finiteCoproduct.ι X a ≫ finiteCoproduct.desc X e = e a := rfl
+
+lemma finiteCoproduct.hom_ext {B : ExtrDisc.{u}} (f g : finiteCoproduct X ⟶ B)
+    (h : ∀ a : α, finiteCoproduct.ι X a ≫ f = finiteCoproduct.ι X a ≫ g) : f = g := by
+  ext ⟨a,x⟩
+  specialize h a
+  apply_fun (fun q => q x) at h
+  exact h
+
+/-- The coproduct cocone associated to the explicit finite coproduct. -/
+@[simps]
+def finiteCoproduct.cocone (F : Discrete α ⥤ ExtrDisc) :
+    Cocone F where
+  pt := finiteCoproduct F.obj
+  ι := Discrete.natTrans fun a => finiteCoproduct.ι F.obj a
+
+/-- The explicit finite coproduct cocone is a colimit cocone. -/
+@[simps]
+def finiteCoproduct.isColimit (F : Discrete α ⥤ ExtrDisc) :
+    IsColimit (finiteCoproduct.cocone F) where
+  desc := fun s => finiteCoproduct.desc _ fun a => s.ι.app a
+  fac := fun s ⟨a⟩ => finiteCoproduct.ι_desc _ _ _
+  uniq := fun s m hm => finiteCoproduct.hom_ext _ _ _ fun a => by
+    specialize hm a
+    ext t
+    apply_fun (fun q => q t) at hm
+    exact hm
+
+/-- The category of extremally disconnected spaces has finite coproducts.
+-/
+instance hasFiniteCoproducts : HasFiniteCoproducts ExtrDisc.{u} where
+  out _ := {
+    has_colimit := fun F => {
+      exists_colimit := ⟨{
+        cocone := finiteCoproduct.cocone F
+        isColimit := finiteCoproduct.isColimit F }⟩ } }
+
+/--
+The coproduct cocone associated to the explicit finite coproduct.
+-/
+@[simps]
+def finiteCoproduct.explicitCocone : Limits.Cocone (Discrete.functor X) where
+  pt := finiteCoproduct X
+  ι := Discrete.natTrans fun ⟨a⟩ => finiteCoproduct.ι X a
+
+/--
+The explicit finite coproduct cocone is a colimit cocone.
+-/
+@[simps]
+def finiteCoproduct.isColimit' : Limits.IsColimit (finiteCoproduct.explicitCocone X) where
+  desc := fun s => finiteCoproduct.desc _ fun a => s.ι.app ⟨a⟩
+  fac := fun s ⟨a⟩ => finiteCoproduct.ι_desc _ _ _
+  uniq := fun s m hm => finiteCoproduct.hom_ext _ _ _ fun a => by
+    specialize hm ⟨a⟩
+    ext t
+    apply_fun (fun q => q t) at hm
+    exact hm
+
+section Iso
+
+noncomputable
+def FromFiniteCoproductIso : finiteCoproduct X ≅ ∐ X :=
+Limits.IsColimit.coconePointUniqueUpToIso
+  (finiteCoproduct.isColimit' X) (Limits.colimit.isColimit _)
+
```

Comment:
Is there a way to avoid having both `isColimit` and `isColimit'` (`cocone` and `explicitCocone`)?